core/controller/src/main/scala/org/apache/openwhisk/core/controller/WebActions.scala

```diff
@@ -553,7 +553,14 @@ trait WhiskWebActionsApi
                                              httpEntity: HttpEntity)(implicit transid: TransactionId) = {
 
     def process(body: Option[JsValue], isRawHttpAction: Boolean) = {
-      processRequest(actionOwnerIdentity, action, extension, onBehalfOf, context.withBody(body), isRawHttpAction)
+      processRequest(
+        actionOwnerIdentity,
+        action,
+        extension,
+        onBehalfOf,
+        context.withBody(body),
+        isRawHttpAction,
+        context.queryAsMap.contains("debug"))
```

Comment:
this captures `debug` as a "system" query parameter and should be documented.. alternatively, why not use a `X-` header (similar to the one for extra system tracing)?

(actions may still use the `debug` query parameter from what I see but this could have unintended consequences.)